unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/UpdateEventHandler/UpdateDispatcher.cs

```diff
@@ -45,10 +63,21 @@ void Awake()
             EnsureEventType(IUpdateEventHandler.EventType.OnDestroy);
         }
 
-        void EnsureEventType( IUpdateEventHandler.EventType eventType )
+        void EnsureEventType(IUpdateEventHandler.EventType eventType)
         {
-            if ( !eventCollections.ContainsKey(eventType) )
+            bool eventTypeExists = false;
+            foreach (var key in eventCollections.Keys)
+            {
+                if (key == eventType)
+                {
+                    eventTypeExists = true;
+                    break;
+                }
+            }
```

Comment:
You can extract a `Contains` method here. It will also allow you to get rid of `eventTypeExsists`.

Reply:
done :)